### PR TITLE
FEAT: Create the room creation menu

### PR DESCRIPTION
**Pull request by Thomaltarix**
I created the room creation scene.

The room looks like this:
image

The user is able to write the name of the room and its player max number.

> [!Note]
> Users can't edit the password text box while the private checkbox isn't checked

The user can choose if he allows cheats or not, privacy of its room or not, etc...

> [!Warning]
> All buttons callbacks linked to the rooms or network aren't done yet (since it's not implement on main yet)

### CODE REVIEW

**Review by Peralban — APPROVED** [review]

File: R-TypeGame/Systems/Menus/Buttons/RoomPasswordTextBox/RoomPasswordTextBoxInitSystem.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        reg.set_component<IComponent>(idxPacketEntities, textLimit, textLimit->getType());
+    }
+
+    std::shared_ptr<DefaultTextComponent> defaultText = parseDefaultText(PATH_JSON);
+    if (defaultText) {
+        reg.register_component<IComponent>(defaultText->getType());
+        reg.set_component<IComponent>(idxPacketEntities, defaultText, defaultText->getType());
+    }
+
+    std::shared_ptr<IComponent> draw = std::make_shared<DrawComponent>();
```
Comment (mathieurobert1) [inline R-TypeGame/Systems/Menus/Buttons/RoomPasswordTextBox/RoomPasswordTextBoxInitSystem.cpp:62]: Check the shared ptr please.

File: R-TypeGame/Systems/Menus/Buttons/RoomPlayerNumberTextBox/RoomPlayerNumberTextBoxInitSystem.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        reg.set_component<IComponent>(idxPacketEntities, textLimit, textLimit->getType());
+    }
+
+    std::shared_ptr<DefaultTextComponent> defaultText = parseDefaultText(PATH_JSON);
+    if (defaultText) {
+        reg.register_component<IComponent>(defaultText->getType());
+        reg.set_component<IComponent>(idxPacketEntities, defaultText, defaultText->getType());
+    }
+
+    std::shared_ptr<IComponent> draw = std::make_shared<DrawComponent>();
```
Comment (mathieurobert1) [inline R-TypeGame/Systems/Menus/Buttons/RoomPlayerNumberTextBox/RoomPlayerNumberTextBoxInitSystem.cpp:62]: Same here.

**Review by mathieurobert1 — CHANGES_REQUESTED** [review]
Just some shared_ptr to check

File: R-TypeGame/Systems/Menus/Buttons/RoomPasswordTextBox/RoomPasswordTextBoxInitSystem.cpp (hunk 1 above)
Comment (Thomaltarix) [reply R-TypeGame/Systems/Menus/Buttons/RoomPasswordTextBox/RoomPasswordTextBoxInitSystem.cpp:62]: We discuss the necessity and usefulness of it, and manage to find that it wasn't

File: R-TypeGame/Systems/Menus/Buttons/RoomPlayerNumberTextBox/RoomPlayerNumberTextBoxInitSystem.cpp (hunk 2 above)
Comment (Thomaltarix) [reply R-TypeGame/Systems/Menus/Buttons/RoomPlayerNumberTextBox/RoomPlayerNumberTextBoxInitSystem.cpp:62]: We discuss the necessity and usefulness of it, and manage to find that it wasn't

**Pull request merged**